test/integration/node-specific/mongo_client.test.ts

```diff
@@ -0,0 +1,605 @@
+
+        expect(findCommandStarted).to.have.property('commandName', 'find');
+        expect(client.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+        expect(client.s.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+
+        // Assertion is redundant but it shows that no initial ping is run
+        expect(findCommandStarted.commandName).to.not.equal('ping');
+
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'automatically connects upon first operation (insertOne)',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const insertOneCommandToBeStarted = once(client, 'commandStarted');
+        await client.db().collection('test').insertOne({ a: 1 });
+        const [insertCommandStarted] = await insertOneCommandToBeStarted;
+
+        expect(insertCommandStarted).to.have.property('commandName', 'insert');
+        expect(client.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+        expect(client.s.options).to.not.have.property(Symbol.for('@@mdb.skipPingOnConnect'));
+
+        // Assertion is redundant but it shows that no initial ping is run
+        expect(insertCommandStarted.commandName).to.not.equal('ping');
+
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'passes connection errors to the user through the first operation',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        const client = this.configuration.newClient(
+          'mongodb://iLoveJavascript?serverSelectionTimeoutMS=100',
+          { monitorCommands: true }
+        );
+
+        const result = await client
+          .db('test')
+          .collection('test')
+          .findOne()
+          .catch(error => error);
+
+        expect(result).to.be.instanceOf(MongoServerSelectionError);
+        expect(client).to.be.instanceOf(MongoClient);
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+      }
+    );
+
+    it(
+      'client.close will not permit operations to auto reconnect',
+      { requires: { auth: 'enabled' } },
+      async function () {
+        await client.db('test').collection('test').findOne();
+
+        expect(client).to.be.instanceOf(MongoClient);
+        expect(client).to.have.property('topology').that.is.instanceOf(Topology);
+
+        await client.close();
+
+        expect(client).to.have.property('topology', undefined);
+
+        const result = await client
+          .db('test')
+          .collection('test')
+          .findOne()
+          .catch(error => error);
+
+        expect(client).to.have.property('topology', undefined);
+        expect(result).to.be.instanceOf(MongoNotConnectedError);
+
+        await client.close();
+      }
+    );
+
+    it(
+      'client.close will not permit operations to auto reconnect permanently',
```

Comment:
there's too much going on in this test; for starters, this is inside an "implicit" reconnect, but it also has an explicit connect in there... and the explicit connect is unclear in terms of what it's supposed to accomplish - do we want to be able to run operations after explicitly calling connect or not? The behavior, I think, is unexpected - what does it mean to connect "successfully" but still not be able to run operations? If we aren't able to run operations after explicit connect, then we should throw an error in the connect itself, not some time later. I'd like to see test cases which clarify behavior of the "close" method in the following cases

Running no operations and no connect to start:
1) create client, immediately call close, try to run an operation
2) create client, immediately call close, explicitly call connect, try to run an operation
Running no operations but with connect to start:
3) create client, explicitly connect, immediately call close, try to run an operation
4) create client, explicitly connect, immediately call close, explicitly call connect, try to run an operation
Running an operation to implicitly connect to start:
5) create client, run an implicitly connecting operation, call close, try to run an operation
6) create client, run an implicitly connecting operation, call close, explicitly call connect, try to run an operation
Running an operation with an explicit connection to start:
7) create client, explicitly connect, run an operation, call close, try to run an operation again
8) create client, explicitly connect, run an operation, call close, explicitly call connect, try to run an operation again

In the above, the odd test cases would be testing the implicit reconnect, and the even cases would be testing the explicit reconnect